java.hints/test/unit/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitTypeTest.java

```diff
@@ -0,0 +1,272 @@
+                .run(ConvertVarToExplicitType.class)
+                .assertNotContainsWarnings(VAR_CONV_DESC);
+    }
+    
+    @Test
+    public void testVarToMethodRetType1() throws Exception {
+        HintTest.create()
+                .setCaretMarker('^')
+                .input("package test;\n"
+                        + "import java.util.ArrayList;\n"
+                        + "public class Test {\n"
+                        + "    public void m() {\n"
+                        + "        var obj = t()^;\n"
+                        + "    }\n"
+                        + "    Object t()\n"
+                        + "    {\n"
+                        + "        return new ArrayList<String>();\n"
+                        + "    }\n"
+                        + "}")
+                .sourceLevel("1.10")
+                .run(ConvertVarToExplicitType.class)
+                .findWarning("4:8-4:22:" + VAR_CONV_WARNING)
+                .applyFix()
+                .assertCompilable()
+                .assertVerbatimOutput("package test;\n"
+                        + "import java.util.ArrayList;\n"
+                        + "public class Test {\n"
+                        + "    public void m() {\n"
+                        + "        Object obj = t();\n"
+                        + "    }\n"
+                        + "    Object t()\n"
+                        + "    {\n"
+                        + "        return new ArrayList<String>();\n"
+                        + "    }\n"
+                        + "}");
+    }
+
+    @Test
+    public void testVarToMethodRetType2() throws Exception {
+        HintTest.create()
+                .setCaretMarker('^')
+                .input("package test;\n"
+                        + "import java.util.Collections;\n"
+                        + "import java.util.List;\n"
+                        + "import java.util.ArrayList;\n"
+                        + "public class Test {\n"
+                        + "    public static void main(String[] args) {\n"
+                        + "        var list = Collections.unmodifiableList(new ArrayList<String>())^;\n"
+                        + "    }\n"
+                        + "}")
+                .sourceLevel("1.10")
+                .run(ConvertVarToExplicitType.class)
+                .findWarning("6:8-6:73:" + VAR_CONV_WARNING)
+                .applyFix()
+                .assertCompilable()
+                .assertVerbatimOutput("package test;\n"
+                        + "import java.util.Collections;\n"
+                        + "import java.util.List;\n"
+                        + "import java.util.ArrayList;\n"
+                        + "public class Test {\n"
+                        + "    public static void main(String[] args) {\n"
+                        + "        List<String> list = Collections.unmodifiableList(new ArrayList<String>());\n"
+                        + "    }\n"
+                        + "}");
+    }
+
```

Comment:
Could you please add a testcase with multiple variable declarators like `var a = 1, b = 2;` ?

Reply:
Compound var is an invalid statement, we get compilation error "'var' is not allowed in a compound declaration". So I could not add test for this as it fails in compilation step. But in Editor, for such statement, multiple 'Convert var to explicit type' hints appear for a compound statement(one for each declaration in compound statement). This is because CompilationUnitTree doesn't contain compound var statement but has separate declaration statements with explicit types.
Clicking on any hint converts all var declarations in compound statement to corresponding statements with explicit type.
var i = 10, j = 20; gets converted to 
int i = 10;
int j = 20;

I think this behaviour is incorrect: 1) there are duplicate hints shown, 2) this hint should have converted statement to int i = 10, j = 20;

For 1) I need to check duplicate hints display can be prevented.
Regarding 2),
2.1) If compound var statement is var v= 10, w = ""; This can't be converted into compound statement w explicit type as types are different. Should we ignore this statement or split the compound statement into different statements.
2.2) We already have bug to track fix split compound var statement, this is WIP 
https://issues.apache.org/jira/browse/NETBEANS-497

Should we show two hints for compound var?
-Convert var to explicit type
-Split compound var statement

For this transaction, I have disabled hint 'Convert var to explicit type' if there are compilation errors in a statement.
If we decide to have both hints, I will log another bug and accordingly work on converting compound decl to explicitly typed in a separate transaction.
Please let me know your view, Thanks